src/components/message/message-card/MessageCardBase.tsx

```diff
@@ -48,166 +49,200 @@ export interface MessageCardBaseProps {
+							<Checkbox
+								id={messageId}
+								checked={!!isExported}
+								onChange={e => {
+									e.stopPropagation();
+									addMessageToExport?.();
+								}}
+							/>
+						) : (
+							<div className={iconsClassName} />
+						)}
+						<div className={sessionClass} style={sessionArrowStyle} />
+						<p
+							className='mc-header__value mc-header__timestamp'
+							title={`Timestamp: ${formattedTimestamp}`}
+							onMouseEnter={hoverMessage}
+							onMouseLeave={unhoverMessage}>
+							{timestamp && formattedTimestamp}
+						</p>
+						<p className='mc-header__value'>{messageIdWithHighlightedSession}</p>
+						<p className='mc-header__value messageType'>{messageType}</p>
 					</div>
-					<div className='mc-header__item' title={messageType && `Name: ${messageType}`}>
-						<span className='mc-header__value messageType'>{messageType}</span>
+				);
+			}
+			return null;
+		};
+
+		return (
+			<div className={rootClass} onClick={addMessageToExport}>
+				<div className='message-card'>
+					<div className='mc__mc-header mc-header'>{renderMessageInfo()}</div>
+					<div className='mc__mc-body mc-body'>
+						{isScreenshotMsg ? (
+							<div className='mc-body__screenshot'>
+								<MessageScreenshotZoom
+									src={
+										typeof bodyBase64 === 'string'
+											? `data:${message.messageType};base64,${message.bodyBase64}`
+											: ''
+									}
+									alt={message.messageId}
+								/>
+							</div>
+						) : (
+							<div className='mc-body__human'>
+								<MessageCardViewTypeRenderer {...messageViewTypeRendererProps} />
+							</div>
+						)}
 					</div>
 				</div>
-			);
-		}
-		return null;
-	};
-
-	return (
-		<div className={rootClass} onClick={addMessageToExport}>
-			{!isEmbedded && isBookmarked && <div className={bookmarkIconClass} />}
-			<div className='message-card'>
-				<div className='mc__mc-header mc-header'>{renderMessageInfo()}</div>
-				<div className='mc__mc-body mc-body'>
-					{isScreenshotMsg ? (
-						<div className='mc-body__screenshot'>
-							<MessageScreenshotZoom
-								src={
-									typeof bodyBase64 === 'string'
-										? `data:${message.messageType};base64,${message.bodyBase64}`
-										: ''
-								}
-								alt={message.messageId}
-							/>
-						</div>
-					) : (
-						<div className='mc-body__human'>
-							<MessageCardViewTypeRenderer {...messageViewTypeRendererProps} />
-						</div>
-					)}
-				</div>
+				<MessageCardTools {...messageCardToolsConfig} />
 			</div>
-			<MessageCardTools {...messageCardToolsConfig} />
-		</div>
-	);
-}
+		);
+	},
+);
+
+MessageCardBase.displayName = 'MessageCardBase';
+export default MessageCardBase;
 
 function calculateHueValue(session: string): number {
 	const hashCode = getHashCode(session);
 
 	return (hashCode % HUE_SEGMENTS_COUNT) * (360 / HUE_SEGMENTS_COUNT);
 }
+
+function getMessageIdWithHighlightedSession(part: string, i: number, arr: string[]) {
+	if (arr.length === 3) {
```

Comment:
you still have to provide keys, right now message markup breaks after message is updated, try to switch export on and pick binary view type and then click the message.

this function returns jsx, maybe we should make it Session component then?

Reply:
Markup breaking still actual

Reply:
Also we have the problem with the selection highlighting in a binary view mode